substrate/frame/scheduler/src/benchmarking.rs

```diff
@@ -134,132 +132,188 @@ fn make_origin<T: Config>(signed: bool) -> <T as Config>::PalletsOrigin {
 	}
 }
 
-benchmarks! {
+#[benchmarks]
+mod benchmarks {
+	use super::*;
+
 	// `service_agendas` when no work is done.
-	service_agendas_base {
-		let now = BlockNumberFor::<T>::from(BLOCK_NUMBER);
+	#[benchmark]
+	fn service_agendas_base() {
+		let now = BLOCK_NUMBER.into();
 		IncompleteSince::<T>::put(now - One::one());
-	}: {
-		Scheduler::<T>::service_agendas(&mut WeightMeter::new(), now, 0);
-	} verify {
+
+		#[block]
+		{
+			Pallet::<T>::service_agendas(&mut WeightMeter::new(), now, 0);
+		}
+
 		assert_eq!(IncompleteSince::<T>::get(), Some(now - One::one()));
 	}
 
 	// `service_agenda` when no work is done.
-	service_agenda_base {
+	#[benchmark]
+	fn service_agenda_base(
+		s: Linear<0, { T::MaxScheduledPerBlock::get() }>,
+	) -> Result<(), BenchmarkError> {
 		let now = BLOCK_NUMBER.into();
-		let s in 0 .. T::MaxScheduledPerBlock::get();
 		fill_schedule::<T>(now, s)?;
 		let mut executed = 0;
-	}: {
-		Scheduler::<T>::service_agenda(&mut WeightMeter::new(), &mut executed, now, now, 0);
-	} verify {
+
+		#[block]
+		{
+			Pallet::<T>::service_agenda(&mut WeightMeter::new(), &mut executed, now, now, 0);
+		}
+
 		assert_eq!(executed, 0);
+
+		Ok(())
 	}
 
 	// `service_task` when the task is a non-periodic, non-named, non-fetched call which is not
 	// dispatched (e.g. due to being overweight).
-	service_task_base {
+	#[benchmark]
+	fn service_task_base() {
 		let now = BLOCK_NUMBER.into();
 		let task = make_task::<T>(false, false, false, None, 0);
 		// prevent any tasks from actually being executed as we only want the surrounding weight.
 		let mut counter = WeightMeter::with_limit(Weight::zero());
-	}: {
-		let result = Scheduler::<T>::service_task(&mut counter, now, now, 0, true, task);
-	} verify {
-		//assert_eq!(result, Ok(()));
+		let _result;
+
+		#[block]
+		{
+			_result = Pallet::<T>::service_task(&mut counter, now, now, 0, true, task);
+		}
+
+		// assert!(_result.is_ok());
```

Comment:
need to address this comment. more commented code like this below. 
also run benchmarks as we did for democracy pallet

Reply:
@AurevoirXavier please open another PR to address this and run the benchmarks before merging. 
See here how to do it: https://github.com/paritytech/polkadot-sdk/pull/6509#issuecomment-2486221769